src/main/scala/org/broadinstitute/dsde/workbench/avram/dataaccess/HttpSamDao.scala

```diff
@@ -0,0 +1,46 @@
+package org.broadinstitute.dsde.workbench.avram.dataaccess
+
+import java.util.logging.Logger
+
+import cats._
+import cats.implicits._
+import com.softwaremill.sttp.{Id, _}
+import io.circe.Decoder.Result
+import io.circe.{DecodingFailure, Json, ParsingFailure}
+import io.circe.generic.auto._
+import io.circe.parser._
+import org.broadinstitute.dsde.workbench.avram.util.{ErrorResponse, RestClient}
+
+class HttpSamDao(samUrl: String) extends SamDao with RestClient {
+  private val log: Logger = Logger.getLogger(getClass.getName)
+
+  override def getUserStatus(token: String): Either[ErrorResponse, SamUserInfoResponse] = {
+    val request = buildAuthenticatedGetRequest(samUrl, "/register/user/v2/self/info", token)
+    for {
+      response <- request.send()
+      content <- response.body                 leftMap responseToErrorResponse(response)
+      json <- parse(content)                   leftMap circeErrorToErrorResponse
```

Comment:
minor nit - this spacing seems strange to me

Reply:
Understood. It's inspired by what Rob did in #12. I'm going to keep it for now but we can reevaluate.